packages/ui/src/signInWithGoogleButton.tsx

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+'use client';
+
+import { useEffect, useState } from 'react';
+import GoogleIcon from '@mui/icons-material/Google';
+import { Avatar, Button, Menu, MenuItem, Typography } from '@mui/material';
+import { signOut } from 'firebase/auth';
+import firebase from 'firebase/compat/app';
+
+import 'firebase/compat/auth';
+
+const firebaseConfig = {
+	apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
+	authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN,
+	projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
+	storageBucket: process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET,
+	messagingSenderId: process.env.NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID,
+	appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID,
+};
+
+if (!firebase.apps.length) {
+	firebase.initializeApp(firebaseConfig);
+}
+
+const auth = firebase.auth();
+
+const SigninWithGoogleButton = () => {
+	const [isSignedIn, setIsSignedIn] = useState(false);
+	const [userPhotoURL, setUserPhotoURL] = useState('');
+	const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+	const [userName, setUserName] = useState('');
+
+	useEffect(() => {
+		const unsubscribe = auth.onAuthStateChanged((user) => {
+			if (user) {
+				setIsSignedIn(true);
+				setUserPhotoURL(user.photoURL || '');
+				setUserName(user.displayName || '');
+			} else {
+				setIsSignedIn(false);
+				setUserPhotoURL('');
+				setUserName('');
+			}
+		});
+
+		return () => unsubscribe();
+	}, []);
```

Comment:
put this under global state management like `Redux` or `Valtio` (very much shill `Valtio`), also I would love to promote this `useEffect` into a custom hook, e.g `useUserProfile` like this:
```typescript
// assuming we did not have Redux or Valtio, then we must store state under useState, with Valtio this would be even more simple/leaner
const useUserProfile = () => {
  const [profile, setProfile] = useState({});

  useEffect(() => {
    const unsubscribe = auth.onAuthStateChanged((user) => {
       setProfile(user);
    });

    return () => unsubscribe();
  }, []);

  return {
    isSignedIn: profile?.displayName, // if onAuthStateChanged not trigger when the  App load/boot then this logic is broken/wrong
    profile,
  };
};
```